### PR TITLE
zephyr: Add NRF54L configuration

### PR DESCRIPTION
**Pull request by michalek-no**
Adds default Kconfig configuration that allows to build MCUboot for NRF54L.
Currently this configuration turns off WDT and FPROTECT, which is TODO to fix.

### CODE REVIEW

File: boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+CONFIG_BOOT_MAX_IMG_SECTORS=512
```
Comment (nordicjm) [inline boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf:5]: What is the sector size? Assuming 4KiB, this means you expect each slot to be 4*512 = 2MiB - this seems very wrong?

File: boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+# Ensure that the qspi driver is disabled by default
+CONFIG_NORDIC_QSPI_NOR=n
+
+# TODO: below are not yet supported and need fixing
+CONFIG_NRF_RTC_TIMER=n
+CONFIG_FPROTECT=n
+CONFIG_BOOT_WATCHDOG_FEED=n
+CONFIG_MULTITHREADING=y
```
Comment (nordicjm) [inline boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf:14]: Why would multithreading be enabled for no reason?

Comment (michalek-no) [reply boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf:14]: soc_flash_nrf_rram uses semaphores which are getting optimized out without multi threading.

Comment (nordicjm) [reply boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf:14]: Why does a flash driver need a semaphore, surely it should be blocking?

**Comment by michalek-no** [comment]
rebased

File: boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+CONFIG_BOOT_MAX_IMG_SECTORS=256
+
+# Ensure that the qspi driver is disabled by default
+CONFIG_NORDIC_QSPI_NOR=n
+
+# TODO: below are not yet supported and need fixing
+CONFIG_NRF_RTC_TIMER=n
+CONFIG_FPROTECT=n
+CONFIG_BOOT_WATCHDOG_FEED=n
+CONFIG_MULTITHREADING=n
```
Comment (nordicjm) [inline boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf:14]:
```suggestion
```

File: boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+CONFIG_BOOT_MAX_IMG_SECTORS=256
+
+# Ensure that the qspi driver is disabled by default
+CONFIG_NORDIC_QSPI_NOR=n
+
+# TODO: below are not yet supported and need fixing
+CONFIG_NRF_RTC_TIMER=n
```
Comment (nordicjm) [inline boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf:11]: Not sure why `CONFIG_NRF_RTC_TIMER` is being touched here, if I remember the PR in zephyr correctly then this Kconfig has been changed to only be selectable if grtc is not selected and grtc should be selected by default?

**Review by nordicjm — APPROVED** [review]

File: boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
```
Comment (nordicjm) [inline boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp.conf:1]: Year is wrong but not that important

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**